src/webui/api/appcontroller.cpp

```diff
@@ -517,3 +517,18 @@ void AppController::defaultSavePathAction()
 {
     setResult(BitTorrent::Session::instance()->defaultSavePath());
 }
+
+void AppController::categoriesAction()
```

Comment:
Move to the TorrentsController.

Reply:
`TorrentsController` contains methods that act on a specific torrent. `categoriesAction`returns a list of all categories that exist for the entire application.

Reply:
>TorrentsController contains methods that act on a specific torrent. 

Oh really?
```c++
class TorrentsController : public APIController
{
    // ...
    void createCategoryAction();
    void editCategoryAction();
    void removeCategoriesAction();
    // ...
```

As I recall, we've already discussed your misunderstanding of the role of AppController...
Maybe it wasn't you, either. But I don't remember the last time someone else contributed in WebAPI code. Have to repeat.
AppController is only intended for a global app-related actions that cannot be distinguished as a separate group (or it doesn't make sense). See into it carefully!
>categoriesAction returns a list of all categories that exist for the entire application.

Everything else also exists for the entire application (e.g. torrents). Let's move all into AppController?

Torrent categories aren't application own property (like, e.g., app version). They belong to BitTorrent subsystem.
Well, I don't have time to talk about it much. The conclusion is that it should be either in TorrentsController near the other category-related actions or in separate TorrentCategoriesController (if there's a good reason for it). I personally would prefer the first option (TorrentsController).

Reply:
Ok, moved to `TorrentsController` and rebased.